File: CS/CustomMessaging.Module.Win/CustomWinFormsMessageBoxMessaging.cs
using System.Windows.Forms;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Win.Core;

namespace CustomMessaging.Module.Win {
    public class CustomWinFormsMessageBoxMessaging : Messaging {
        public CustomWinFormsMessageBoxMessaging(XafApplication application)
            : base(application) { }
        protected override DialogResult ShowCore(
            string message, string caption, MessageBoxButtons buttons, MessageBoxIcon icon) {            
            return MessageBox.Show(message, caption, buttons, icon);
        }
    }
}

File: VB/CustomMessaging.Module.Win/CustomWinFormsMessageBoxMessaging.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Win.Core

Namespace CustomMessaging.Module.Win
    Public Class CustomWinFormsMessageBoxMessaging
        Inherits Messaging
        Public Sub New(ByVal application As XafApplication)
            MyBase.New(application)
        End Sub
        Protected Overloads Overrides Function ShowCore(ByVal message As String, ByVal caption As String, ByVal buttons As MessageBoxButtons, ByVal icon As MessageBoxIcon) As DialogResult
            Dim result As DialogResult = MessageBox.Show(message, caption, buttons, icon)
            Return result
        End Function
    End Class
End Namespace
